py/dml/serialize.py

```diff
@@ -372,80 +393,158 @@ def type_signature(dmltype, is_for_serialization):
                       if not is_for_serialization else '')
     assert False
 
+def serialize_sources_to_list(site, sources, out_attr):
```

Comment:
naming: the word `sources` is kind of redundant, can be removed (or replaced if you prefer).

Reply:
I don't really like the terseness of `serialize_list`... I feel like it's too easy to confuse it with talking about a specific serialized datatype, while this function is for a much more generic purpose than that. If you insist upon it though...